src/greedycd.jl

```diff
@@ -51,12 +51,12 @@ mutable struct GreedyCDUpd_State{T}
     SH::Matrix{T}
     DW::Matrix{T}
     DH::Matrix{T}
-    qW::Array{Int}
-    qH::Array{Int}
+    qW::Vector{Int}
+    qH::Vector{Int}
     
     function GreedyCDUpd_State{T}(X, W::Matrix{T}, H::Matrix{T}) where T
         p, n, k = nmf_checksize(X, W, H)
-        new{T}(W * H, 
+        new{T}(Matrix{T}(undef, p, n),
```

Comment:
Why change this bit? I think the `WH` field of the type is intended to be the product of `W` and `H`.

Reply:
In order to calculate the product of W and H in the `evaluate_objv`, I've removed the duplicate calculation.